server/lib/api/matchmakingTimes.ts

```diff
@@ -0,0 +1,173 @@
+import Koa from 'koa'
+import Router from '@koa/router'
+import httpErrors from 'http-errors'
+import Joi from 'joi'
+
+import { checkAllPermissions } from '../permissions/check-permissions'
+import ensureLoggedIn from '../session/ensure-logged-in'
+import { MATCHMAKING_TYPES } from '../../../common/constants'
+import { MATCHMAKING } from '../../../common/flags'
+import { AddMatchmakingTimeBody } from '../../../common/matchmaking'
+import { featureEnabled } from '../flags/feature-enabled'
+import { joiValidator } from '../validation/joi-validator'
+import {
+  getCurrentMatchmakingTime,
+  getFutureMatchmakingTimes,
+  getPastMatchmakingTimes,
+  addMatchmakingTime,
+  getMatchmakingTimeById,
+  removeMatchmakingTime,
+} from '../models/matchmaking-times'
+
+const matchmakingTypeSchema = Joi.object({
+  matchmakingType: Joi.valid(...MATCHMAKING_TYPES).required(),
+})
+
+const addMatchmakingTimeSchema = Joi.object({
+  startDate: Joi.number().integer().greater(Date.now()).required(),
```

Comment:
I don't think this validation doesn't what you want it to. `Date.now` will be evaluated once, when this file is evaluated, and never again for the duration of the process. You probably want to use the dynamic validation stuff Joi has instead